pytorch_lightning/loops/batch/training_batch_loop.py

```diff
@@ -498,13 +289,6 @@ def backward(
             opt_idx: Index of the current optimizer being used. ``None`` if using manual optimization.
         """
         self.trainer.accelerator.backward(loss, optimizer, opt_idx, *args, **kwargs)
-
-        if not self.trainer.fit_loop.should_accumulate():
```

Comment:
reviewer: this removed block is all specific to automatic optimization and gets moved to the OptimizerLoop